filters/shedder/admission.go

```diff
@@ -0,0 +1,366 @@
+// aggression >= 1, 1: linear, 2: qudratic, .., see also https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/admission_control_filter#aggression
+//
+// see also https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/admission_control_filter#admission-control
+func (spec *admissionControlSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var err error
+
+	metricSuffix := ""
+	if len(args) > 0 {
+		ok := false
+		metricSuffix, ok = args[0].(string)
+		if !ok {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	mode := inactive
+	if len(args) > 1 {
+		mode, err = getModeArg(args[1])
+		if err != nil {
+			logrus.Warnf("mode failed: %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	d := defaultDuration
+	if len(args) > 2 {
+		d, err = getDurationArg(args[2])
+		if err != nil {
+			logrus.Warnf("d failed: %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	windowSize := defaultWindowSize
+	if len(args) > 3 {
+		windowSize, err = getIntArg(args[3])
+		if err != nil {
+			logrus.Warnf("windowsize failed: %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+		if windowSize < minWindowSize {
+			windowSize = minWindowSize
+		}
+		if windowSize > maxWindowSize {
+			windowSize = maxWindowSize
+		}
+	}
+
+	minRequests := defaultMinRps
+	if len(args) > 4 {
+		minRequests, err = getIntArg(args[4])
+		if err != nil {
+			logrus.Warnf("minRequests failed: %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	threshold := defaultThreshold
+	if len(args) > 5 {
+		threshold, err = getFloat64Arg(args[5])
+		if err != nil {
+			logrus.Warnf("threshold failed %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	maxRejectProbability := defaultMaxRejectProbability
+	if len(args) > 6 {
+		maxRejectProbability, err = getFloat64Arg(args[6])
+		if err != nil {
+			logrus.Warnf("maxRejectProbability failed: %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	aggression := defaultAggression
+	if len(args) > 7 {
+		aggression, err = getFloat64Arg(args[7])
+		if err != nil {
+			logrus.Warnf("aggression failed: %v", err)
+			return nil, filters.ErrInvalidFilterParameters
+		}
+		if aggression <= 0.0 {
+			logrus.Warn("aggression should be >0")
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	}
+
+	ac := &admissionControl{
+		metrics:      metrics.Default,
+		metricSuffix: metricSuffix,
+		tracer:       spec.tracer,
+
+		mode:                 mode,
+		d:                    d,
+		windowSize:           windowSize,
+		minRequests:          minRequests,
+		successThreshold:     threshold,
+		maxRejectProbability: maxRejectProbability,
+		aggression:           aggression,
+
+		totals:         make([]int64, windowSize),
+		success:        make([]int64, windowSize),
+		counter:        atomic.NewInt64(0),
+		successCounter: atomic.NewInt64(0),
+	}
+	go ac.tickWindows()
```

Comment:
Hm, this will leak goroutines.

Reply:
Hmm, I thought you would over look this :D
fixed

Reply:
This is now fixed via PostProcessor handling

Reply:
This was fixed